### DISCUSSION TITLE
Change filter ban argument

### DISCUSSION THREAD

**Discussion by StefanDumitrache** [fail2ban/fail2ban#3326] [General]
Hello, 

I have this filter:
```
[Definition]

prefregex = ^\s*\S+\s*ppt?pd\[<F-MLFID>\d+</F-MLFID>\]: CTRL: <F-CONTENT>(?:Client|PTY)\b.+</F-CONTENT>$

failregex = ^PTY read or GRE write failed
            ^Client <ADDR> control connection <F-NOFAIL>started</F-NOFAIL>$
```

I want to change this filter to ban when only the password is wrong because I saw that when users log out for various reasons, still appears: CTRL: PTY0 read or GRE write failed (pty,gre)=(6,7) and bans the user when he disconnects or the internet fails when connecting.

I want the new filter to ban when it appears: Peer failed CHAP authentication.

Below you will find logs when the password is wrong and when the user has disconnected due to the internet.

Log when the password is wrong:
```
Jul 29 14:34:39 vpn-nou pptpd[787667]: CTRL: Client 213.233.110.96 control connection started
Jul 29 14:34:39 vpn-nou pptpd[787667]: CTRL: Starting call (launching pppd, opening GRE)
Jul 29 14:34:39 vpn-nou pppd[787668]: Plugin /usr/lib/pptpd/pptpd-logwtmp.so loaded.
Jul 29 14:34:39 vpn-nou pppd[787668]: pppd 2.4.9 started by root, uid 0
Jul 29 14:34:39 vpn-nou pppd[787668]: Using interface ppp6
Jul 29 14:34:39 vpn-nou pppd[787668]: Connect: ppp6 <--> /dev/pts/10
Jul 29 14:34:39 vpn-nou NetworkManager[412]: <info>  [1659094479.3673] manager: (ppp6): new Ppp device (/org/freedesk>
Jul 29 14:34:39 vpn-nou systemd-udevd[787669]: ethtool: autonegotiation is unset or enabled, the speed and duplex are>
Jul 29 14:34:39 vpn-nou pppd[787668]: Peer stefan.dumitrache failed CHAP authentication
Jul 29 14:34:39 vpn-nou pppd[787668]: Connection terminated.
Jul 29 14:34:39 vpn-nou pppd[787668]: Exit.
Jul 29 14:34:39 vpn-nou gnome-shell[605]: Removing a network device that was not added
Jul 29 14:34:39 vpn-nou pptpd[787667]: GRE: read(fd=6,buffer=55bb3cab2520,len=8196) from PTY failed: status = -1 erro>
Jul 29 14:34:39 vpn-nou pptpd[787667]: CTRL: PTY read or GRE write failed (pty,gre)=(6,7)
Jul 29 14:34:39 vpn-nou pptpd[787667]: CTRL: Reaping child PPP[787668]
Jul 29 14:34:39 vpn-nou pptpd[787667]: CTRL: Client 213.233.110.96 control connection finished
```

Log when the user's internet connection failed:
```
Jul 29 14:26:00 vpn-nou pptpd[787400]: CTRL: Client 213.233.110.96 control connection started
Jul 29 14:26:00 vpn-nou pptpd[787400]: CTRL: Starting call (launching pppd, opening GRE)
Jul 29 14:26:00 vpn-nou pppd[787401]: Plugin /usr/lib/pptpd/pptpd-logwtmp.so loaded.
Jul 29 14:26:00 vpn-nou pppd[787401]: pppd 2.4.9 started by root, uid 0
Jul 29 14:26:00 vpn-nou pppd[787401]: Using interface ppp6
Jul 29 14:26:00 vpn-nou pppd[787401]: Connect: ppp6 <--> /dev/pts/10
Jul 29 14:26:00 vpn-nou systemd-udevd[787402]: ethtool: autonegotiation is unset or enabled, the speed and duplex are>
Jul 29 14:26:00 vpn-nou NetworkManager[412]: <info>  [1659093960.5265] manager: (ppp6): new Ppp device (/org/freedesk>
Jul 29 14:26:30 vpn-nou pppd[787401]: LCP: timeout sending Config-Requests
Jul 29 14:26:30 vpn-nou pppd[787401]: Connection terminated.
Jul 29 14:26:30 vpn-nou gnome-shell[605]: Removing a network device that was not added
Jul 29 14:26:30 vpn-nou pppd[787401]: Modem hangup
Jul 29 14:26:30 vpn-nou pppd[787401]: Exit.
Jul 29 14:26:30 vpn-nou pptpd[787400]: GRE: read(fd=6,buffer=55625788d520,len=8196) from PTY failed: status = -1 erro>
Jul 29 14:26:30 vpn-nou pptpd[787400]: CTRL: PTY read or GRE write failed (pty,gre)=(6,7)
Jul 29 14:26:30 vpn-nou pptpd[787400]: CTRL: Reaping child PPP[787401]
Jul 29 14:26:30 vpn-nou pptpd[787400]: CTRL: Client 213.233.110.96 control connection finished
```

Please help us with this problem, thank you!

**Comment by sebres** [comment]
It is impossible at the moment - the issue with that logging is that the messages with the session-ID providing the IP address (787667 in first case and 787400 in the second case) are completely equivalent in both cases.
The message `pppd[787668]: Peer ... failed CHAP authentication` is throwing in context of other "session-ID" (787668, not 787667), that cannot be referenced with 787667 (and therefore with the IP address), because fail2ban has no such functionality at the moment.

I told you in https://github.com/fail2ban/fail2ban/issues/3276#issuecomment-1126169893 that the filter would only work:
*If it is always the case on every failed attempt and never the case on succeeded attempts*
Unfortunately you have found now a "success" case where it is not so.

So there is really no way, unless fail2ban doesn't implement such kind of mapping by `pptpd[787667]: CTRL: Reaping child PPP[787668]` to combine the "sessions"  messages of 787668 (which authenticate) to 787667 (which hold the IP address) or this VPN service changes its logging behavior.
You can try to increase `maxretry` (or decrease `findtime`), but you could anyway get false positives further for the clients with unstable connections.

**Comment by StefanDumitrache** [comment]
Thank you very much for your reply, have a nice day!
